frontend/packages/knative-plugin/src/components/add/event-sources/SinkSection.tsx

```diff
@@ -1,87 +1,45 @@
-  modelFor,
-  referenceFor,
-  K8sResourceKind,
-} from '@console/internal/module/k8s';
+import { getFieldId, ResourceDropdownField } from '@console/shared';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-
-type DropdownItemProps = {
-  model: K8sKind;
-  name: string;
-};
-
-const DropdownItem: React.FC<DropdownItemProps> = ({ model, name }) => (
-  <span className="co-resource-item">
-    <span className="co-resource-icon--fixed-width">
-      <ResourceIcon kind={referenceForModel(model)} />
-    </span>
-    <span className="co-resource-item__resource-name">
-      <span className="co-resource-item__resource-api co-truncate show co-nowrap small">
-        {name}
-      </span>
-    </span>
-  </span>
-);
+import { knativeServingResourcesServices } from '../../../utils/create-knative-utils';
 
 export interface SinkSectionProps {
-  services: K8sResourceKind[];
+  namespace: string;
 }
 
-const SinkSection: React.FC<SinkSectionProps> = ({ services }) => {
+const SinkSection: React.FC<SinkSectionProps> = ({ namespace }) => {
   const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
   const autocompleteFilter = (strText, item): boolean => fuzzy(strText, item?.props?.name);
   const fieldId = getFieldId('sink-name', 'dropdown');
-  const allItems = _.reduce(
-    services,
-    (acc, itemData) => {
-      const svcName = _.get(itemData, 'metadata.name');
-      return {
-        ...acc,
-        [svcName]: (
-          <DropdownItem
-            key={itemData.metadata.uid}
-            model={modelFor(referenceFor(itemData))}
-            name={itemData.metadata.name}
-          />
-        ),
-      };
+  const onChange = React.useCallback(
+    (selectedValue) => {
+      if (selectedValue) {
+        setFieldTouched('sink.knativeService', true);
+        setFieldValue('sink.knativeService', selectedValue);
+        validateForm();
+      }
     },
-    {},
+    [setFieldValue, setFieldTouched, validateForm],
   );
-
-  const onChange = (selectedValue) => {
-    if (selectedValue) {
-      setFieldTouched('sink.knativeService', true);
-      setFieldValue('sink.knativeService', selectedValue);
-      validateForm();
-    }
-  };
-
   return (
     <FormSection title="Sink">
-      <FormGroup
-        fieldId={fieldId}
-        label="Knative Service"
-        helperText="Select a Service to sink to."
-        isRequired
-      >
-        <Dropdown
-          autocompleteFilter={autocompleteFilter}
-          items={allItems}
-          dropDownClassName="dropdown--full-width"
-          menuClassName="dropdown-menu--text-wrap"
+      <FormGroup fieldId={fieldId} helperText="Select a Service to sink to." isRequired>
+        <ResourceDropdownField
+          name="sink.knativeService"
+          label="Knative Service"
+          resources={knativeServingResourcesServices(namespace)}
+          dataSelector={['metadata', 'name']}
+          key={namespace}
```

Comment:
Do you specifically need a `key` to force unmounting here? Otherwise remove the key.